src/libambit/device_driver_ambit3.c

```diff
@@ -579,7 +579,10 @@ static int parse_log_header(libambit_sbem0102_data_t *reply_data_object, ambit3_
         data += 19;
         break;
       case 0x8a: /* gen2+gen3 fw */
-        data += 1; // Skip size ??
+        if (fw_gen == AMBIT3_FW_GEN3)
+            data += 1; // Skip size
```

Comment:
Hmm.. I understand where you want to go with this, and I agree it is the correct way. But my guess is it will break the parse_log_header for AMBIT_FW_GEN3, because the src/libambit/device_driver_ambit3.c is not perfect.

The 0x8a type starts with:
8aff 7e03 0000 [Here is the date/time]

libambit_sbem0102_data_ptr(data) -> returns data[2]:
7e03 0000 [Here is the date/time]

So we skip first 4 bytes to get to date/time.

And to know where to end the log_head we use some super magic *ironic* code.*
It is not perfect, but it works without rewriting sbem0102.

I want you to reverse commit  	3e94e19 to d14b874, not because they are wrong, but it will need a total rewrite of the parser code to be usefull. (as of my understanding of the device_driver_ambit3.c code)